src/index.js

```diff
@@ -93,81 +96,38 @@ const PLUGIN_NAME = 'critters-webpack-plugin';
  */
 
-/**
- * Create a Critters plugin instance with the given options.
- * @public
- * @param {Options} options Options to control how Critters inlines CSS.
- * @example
- * // webpack.config.js
- * module.exports = {
- *   plugins: [
- *     new Critters({
- *       // Outputs: <link rel="preload" onload="this.rel='stylesheet'">
- *       preload: 'swap',
- *
- *       // Don't inline critical font-face rules, but preload the font URLs:
- *       preloadFonts: true
- *     })
- *   ]
- * }
- */
 export default class Critters {
   /** @private */
-  constructor (options) {
-    this.options = Object.assign({ logLevel: 'info', externalStylesheets: [] }, options || {});
-    this.options.pruneSource = this.options.pruneSource !== false;
+  constructor(options) {
+    this.options = Object.assign(
+      {
+        logLevel: 'info',
+        path: '',
+        publicPath: '',
+        reduceInlineStyles: true,
+        pruneSource: false,
+        additionalStylesheets: [],
+        ssrMode: false
+      },
+      options || {}
+    );
+
     this.urlFilter = this.options.filter;
     if (this.urlFilter instanceof RegExp) {
       this.urlFilter = this.urlFilter.test.bind(this.urlFilter);
     }
-    this.logger = log({ name: 'Critters', unique: true, level: this.options.logLevel });
-  }
 
-  /**
-   * Invoked by Webpack during plugin initialization
-   */
-  apply (compiler) {
-    // hook into the compiler to get a Compilation instance...
-    tap(compiler, 'compilation', PLUGIN_NAME, false, compilation => {
-      // ... which is how we get an "after" hook into html-webpack-plugin's HTML generation.
-      if (compilation.hooks && compilation.hooks.htmlWebpackPluginAfterHtmlProcessing) {
-        tap(compilation, 'html-webpack-plugin-after-html-processing', PLUGIN_NAME, true, (htmlPluginData, callback) => {
-          this.process(compiler, compilation, htmlPluginData.html)
-            .then(html => { callback(null, { html }); })
-            .catch(callback);
-        });
-      } else {
-        // If html-webpack-plugin isn't used, process the first HTML asset as an optimize step
-        tap(compilation, 'optimize-assets', PLUGIN_NAME, true, (assets, callback) => {
-          let htmlAssetName;
-          for (const name in assets) {
-            if (name.match(/\.html$/)) {
-              htmlAssetName = name;
-              break;
-            }
-          }
-          if (!htmlAssetName) return callback(Error('Could not find HTML asset.'));
-          const html = assets[htmlAssetName].source();
-          if (!html) return callback(Error('Empty HTML asset.'));
-
-          this.process(compiler, compilation, String(html))
-            .then(html => {
-              assets[htmlAssetName] = new sources.RawSource(html);
-              callback();
-            })
-            .catch(callback);
-        });
-      }
-    });
+    this.logger = this.options.logger || logger;
```

Comment:
This no longer respects `logLevel` in the default logger.

Reply:
Fixed this. Please use version `critters@0.0.9` to get this feature